search.go

```diff
@@ -148,9 +148,16 @@ func (e SearchService) PartialExec(idx, statement string, params map[string]inte
 
 	for start := res.Start; start+inc <= total; start += inc {
 		query.Start = start + inc
+
 		fullUrl = fmt.Sprintf("search/%s", query)
 
-		err = e.client.magicRequestDecoder("POST", fullUrl, body, &paged_res)
+		newbody, error := JSONReader(params)
```

Comment:
I'm not sure what changed here. It looks like `newbody` should be the same as `body` from line 132. Can you help me understand?

Reply:
Line 154 builds a new reader every time it is used.  The readers are set up as read once when used and then return empty on subsequent use.  So if body from line 132 is passed to the POST call it looks like no body was specified.

Reply:
Ahhh, gotcha - thank you!

Reply:
Upstream I reset the reader to start at the beginning instead of creating a new reader. I assumed "the chef people" would pick on me for doing something crudely if I just recreated the reader.  :-) It really doesn't matter which gets picked in my opinion.

Reply:
Thanks @MarkGibbons for explaining that. I saw your message about your solution after implementing this fix. We urgently needed this functionality for Chef Analyze reports, we can reconcile when we merge both code bases.